### ISSUE TITLE
Normalizing certain linestrings with repeated points changes their is_simple status

### ISSUE THREAD

**Issue by ferranmesas** [shapely/shapely#1343]
Good afternoon, I'm seeing some unexpected behaviour when checking `is_simple` on a LineString.
When I create, is_simple returns False. but if I normalize it before checking the flag, the points reorder themselves to and then is_simple returns True. 

runnable example:
```python
from shapely.geometry import LineString

l = LineString([(2, 33), (1, 34), (3, 31), (2, 33), (2, 33)])
print(l.is_simple) # prints False
print(l.normalize().is_simple) # prints True
```

I'm aware there's a repeated point in the LineString, and after checking some other issues (#768, #1147) and the docs [1] it's still unclear to me if, even if it's allowed, it breaks the simplicity rules, but I think normalizing the geometry shouldn't change the value.

The coordinates after normalization are `[(1.0, 34.0), (2.0, 33.0), (2.0, 33.0), (3.0, 31.0), (1.0, 34.0)]`, moving the repeated points to the middle of the array.

This is happening in version `1.8.1.post1`. `1.7.1` doesn't have `normalize` as it was introduced later, but reports the original LineString as simple. I haven't tested any other versions.

[1]: [Repeated points in the ordered sequence are allowed, but may incur performance penalties and should be avoided.](https://shapely.readthedocs.io/en/stable/manual.html#LineString)

**Comment by mwtoews** [comment]
Whether the closed `LINESTRING (2 33, 1 34, 3 31, 2 33, 2 33)` is simple or not unfortunately depends on the version of things. This is what I can see:

- Before GEOS 3.10 (e.g. shapely 1.8.0 has GEOS 3.9.1): True (simple)
- After GEOS 3.10 (e.g. shapely 1.8.1 has GEOS 3.10.2): False (non-simple)
- JTS 1.18.3: False, with a comment "Non-simple intersection at POINT ( 2 33 )"

Newer versions are probably correct that this closed line is non-simple.

As for the result of normalize, JTS and both GEOS 3.9.1 and 3.10.2 have the same simple result: `LINESTRING (1 34, 2 33, 2 33, 3 31, 1 34)`. Oddly PostGIS 3's `ST_Normalize()` function returns the same non-simple geometry as the input (I don't know if that function uses GEOS 3.8.0). Not sure if @dr-jts has anything further to add.

**Comment by dr-jts** [comment]
This is a bug.  Closed lines are in fact simple.  This is shown by running `isSimple` on the closed line with the repeated point removed, which gives `true`.  (Also note that LinearRings are closed rings, and must be simple).

I'll log this as a JTS issue and start working on a fix (which will wind up in GEOS too).

I'm also surprised that `normalize` doesn't remove repeated points - this seems like a design flaw, if intentional.

**Comment by dr-jts** [comment]
This is [fixed in GEOS 3.10](https://github.com/libgeos/geos/commit/c7e3624cd34933017091c2960f8081cb3b4da1ac).

**Comment by jorisvandenbossche** [comment]
Thanks @dr-jts! Closing here then.
